.github/workflows/manifest-checks.yaml

```
@@ -0,0 +1,135 @@
+name: Check manifests
+on:
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  # run `kustomize build` over changed kustomize directories
+  # powered by https://github.com/utilitywarehouse/dev-enablement-mono/blob/main/cmd/kustomize-build-dirs/main.go
+  # reach out on #help-dev-enablement if you encounter any issues
+  run-kustomize-build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          # Fetch both the merge commit GitHub generates for the pull request
+          # and both its parents, i.e. the tip of the branch and the tip of main
+          fetch-depth: 2
+      - name: install `kustomize-build-dirs`
+        run: |
+          curl \
+            --location \
+            --silent \
+            https://github.com/utilitywarehouse/manifest-checkers/releases/download/v0.1.0/manifest-checkers_Linux_x86_64.tar.gz \
+          | tar \
+            --directory /usr/local/bin \
+            --extract \
+            --gzip \
+            --file - \
+            kustomize-build-dirs \
+      - name: Ensure Kustomize
+        run: >-
+          command -v kustomize ||
+          curl --silent --location https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv5.2.1/kustomize_v5.2.1_linux_amd64.tar.gz |
+          tar zx -C /usr/local/bin
+      - name: Run `kustomize build` across changed directories
+        run: |
+          set -o pipefail
+          mkdir built-manifests/
+          git ls-files -- '*/kustomization.yaml' | xargs kustomize-build-dirs --out-dir built-manifests/ --truncate-secrets 2>&1 \
+            | tee build-output
```

Can be simplified since we're not doing any diffing logic

```suggestion
        run: >-
          git ls-files -- '*/kustomization.yaml'
          | xargs kustomize-build-dirs --out-dir built-manifests/ --truncate-secrets 2>&1
          | tee build-output
```

---

I'm not sure what do you mean- aren't we doing diffing logic [here](https://github.com/utilitywarehouse/shared-kustomize-bases/blob/DENA-489/.github/workflows/manifest-checks.yaml#L78)?

---

> I'm not sure what do you mean- aren't we doing diffing logic [here](https://github.com/utilitywarehouse/shared-kustomize-bases/blob/DENA-489/.github/workflows/manifest-checks.yaml#L78)?

Yeah, sorry I mean: since we're always building for all files, and not just the ones changed in this PR

---

Anyway- isn't `built-manifests` directory needed?

---

> Anyway- isn't `built-manifests` directory needed?

`kustomize-build-dirs` will created it if it doesn't exist

---

[Applied](https://github.com/utilitywarehouse/shared-kustomize-bases/pull/30/commits/ed79caf0b248f988d819743788db25b0f1f71dce)